### ISSUE TITLE
FEATURE/ Table component support

### ISSUE THREAD

**Issue by kasprzyk-sz** [visual-framework/vf-wp#299]
Table block needs integrations with 
[https://visual-framework.github.io/vf-core/components/detail/vf-table.html](url)

Currently:

Input:
![Screenshot 2020-05-15 at 15 13 33](https://user-images.githubusercontent.com/54020899/82055464-d2a2e700-96c0-11ea-8ed2-0ca9749a06c9.png)

Output:
![Screenshot 2020-05-15 at 15 13 41](https://user-images.githubusercontent.com/54020899/82055491-ddf61280-96c0-11ea-81c2-e08278d2cf8c.png)

**Comment by khawkins98** [comment]
I think this might be technically correct according to the VF: `vf-table` doesn't grow to be full width. 

But maybe `.vf-content .vf-table` should...

**Comment by kasprzyk-sz** [comment]
Anyway, I guess it's a bit outdated.

HTML output:
![Screenshot 2020-05-15 at 15 24 22](https://user-images.githubusercontent.com/54020899/82078461-6be1f580-96e1-11ea-86b1-f97f6fc41bc8.png)

**Comment by khawkins98** [comment]
VF-WP uses the `vf-content` implementation of `vf-table` (so no classes on the table). https://visual-framework.github.io/vf-core/components/detail/vf-content.html

The logic for this is that it's not pragmatic to add all the classes needed to a table.

**Comment by kasprzyk-sz** [comment]
Right! What I meant was table styles. There is only `default` and `stripes` available, and they affect the look only in the editor. Also, there are color settings with no effect on the front end.

As for the full width, maybe a good solution would be to have a toggle switch in table settings to determine it.

**Comment by dbushell** [comment]
It's possible with JavaScript to edit the native Gutenberg table block and add custom styles / options (to some extent).

For the HTML render I've been using PHP filters to replace any markup and classes as necessary.
https://github.com/visual-framework/vf-wp/tree/master/wp-content/plugins/vf-gutenberg/includes/core

There isn't any table block filters yet so it's using the default WP styles.

**Comment by dbushell** [comment]
I will add that of course. I don't think `vf-core` had finalised the table pattern when I last looked at default blocks.

**Comment by sturobson** [comment]
here's what the table looks like inside of `vf-content` as a default 

https://visual-framework.github.io/vf-core/components/detail/vf-content.html

**Comment by sturobson** [comment]
There's also the `vf-table` component - https://visual-framework.github.io/vf-core/components/detail/vf-table.html - but I wouldn't think all of that functionality would make it into a blogpost … let alone let content creators have a choice … they've a choice of headings and that does the semantics of the page no favours.
